ToolLibrary.Standart/DoubleLinkedList.cs

```diff
@@ -0,0 +1,330 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ToolLibrary.Standart
+{
+    /// <summary>
+    /// Represents a node in a <see cref="DoubleLinkedList{T}"/>.
+    /// </summary>
+    /// <typeparam name="T">Specifies the value type.</typeparam>
+    public class DoubleLinkedListNode<T>
+    {
+        internal DoubleLinkedListNode(T value)
+        {
+            Value = value;
+        }
+
+        /// <summary>
+        /// Reference to the previous node.
+        /// </summary>
+        public DoubleLinkedListNode<T> Prev { get; internal set; }
+
+        /// <summary>
+        /// Reference to the next node.
+        /// </summary>
+        public DoubleLinkedListNode<T> Next { get; internal set; }
+  
+        /// <value>Get the value of node.</value>
+        public T Value { get; }
+    }
+
+    
+    /// <summary>
+    /// Represents a doubly linked list.
+    /// </summary>
+    /// <typeparam name="T"> Specifies the element type of the linked list.</typeparam>
+    public class DoubleLinkedList<T> : IEnumerable<DoubleLinkedListNode<T>>
+    {
+        ///<value>Get the first node.</value>
+        public DoubleLinkedListNode<T> FirstNode { get; private set; }
+
+        ///<value>Get the last node.</value>
+        public DoubleLinkedListNode<T> LastNode { get; private set; }
+
+        ///<value>Get count of nodes.</value>
+        public int Count { get; private set; } = 0;
+
+      
+        public DoubleLinkedListNode<T> this[int i]
```

Comment:
This accessor should also needs public comments. In fact, all public methods need to have comments.